app/components/EditProfileForm.tsx

```diff
@@ -1,17 +1,58 @@
 'use client'
 import { FormEvent } from 'react';
+import { useRouter } from 'next/navigation';
 
-export default function EditProfileForm() {
+export default function EditProfileForm({userData}) {
 
-  /**
-   * Submits new post data
-   * 
-   * @param e - the form submit event
-  */
-  const onSubmitHandler = (e: FormEvent<HTMLFormElement>): void => {
+  const router = useRouter();
+
+  const firstName = userData['data'][0]['firstName'][0]
+  const lastName = userData['data'][0]['lastName'][0]
+  const username = userData['data'][0]['username'][0]
+  const gender = userData['data'][0]['gender'][0]
+  const bio = userData['data'][0]['bio'][0]
+
+  const userInfo = new Map([
+    ['firstName', firstName],
+    ['lastName', lastName],
+    ['username', username],
+    ['gender', gender],
+    ['bio', bio]
+  ])
+
+  const onSubmitHandler = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     const formData: Object = Object.fromEntries((new FormData(e.currentTarget)));
-    console.log(JSON.stringify(formData));
+    console.log(formData);
+
+    for (const [key, value] of Object.entries(formData)) {
+      // No support for profile picture yet
```

Comment:
Just so VSCode TODO extension picks this up.
```suggestion
      // TODO: Add support for profile picture
```